addons/Nostrum/prims.lua

```diff
@@ -0,0 +1,433 @@
+    return meta[t].width, meta[t].height
+end
+
+function prims.color(t, red, green, blue)
+    if not blue then
+        return unpack(meta[t].color,2,4)
+    end
+    
+    local m = meta[t]
+    windower.prim.set_color(m.name, m.color[1], red, green, blue)
+    m.color[2], m.color[3], m.color[4] = red, green, blue
+end
+
+function prims.alpha(t, alpha)
+    if not alpha then
+        return meta[t].color[1]
+    end
+    local m = meta[t]
+    windower.prim.set_color(m.name, alpha, m.color[2], m.color[3], m.color[4])
+    m.color[1] = alpha
+end
+
+function prims.argb(t, a, r, g, b)
+    if not b then
+        return unpack(meta[t].color)
+    end
+    
+    local m = meta[t]
+    windower.prim.set_color(m.name, a, r, g, b)
+    m.color[1], m.color[2], m.color[3], m.color[4] = a, r, g, b
+end
+-- Sets/returns prim transparency. Based on percentage values, with 1 being fully transparent, while 0 is fully opaque.
+function prims.transparency(t, alpha)
+    if not alpha then
+        return 1 - meta[t].color[1]/255
+    end
+
+    alpha = math.floor(255*(1-alpha))
+    local m = meta[t]
+    windower.prim.set_color(m.name, alpha, m.color[2], m.color[3], m.color[4])
+    m.color[1] = alpha
+end
+
+function prims.tile(t, x_rep, y_rep)
+    if not y_rep then
+        return unpack(meta[t].tile)
+    end
+    
+    local m = meta[t]
+    windower.prim.set_repeat(m.name, x_rep, y_rep)
+    m.tile[1], m.tile[2] = x_rep, y_rep
+end
+
+function prims.texture(t, path)
+    if not path then
+        return meta[t].texture
+    end
+    
+    windower.prim.set_texture(meta[t].name, path)
+    meta[t].texture = path
+end
+
+function prims.fit(t, fit)
+    if fit == nil then
+        return meta[t].fit
+    end
+    
+    windower.prim.set_fit_to_texture(meta[t].name, fit)
+    meta[t].fit = fit
+end
+
+function prims.hover(t, x, y)
+    local m = meta[t]
+    return (m.x2 >= x
+        and m.x1 <= x
+        and m.y1 >= y
+        and m.y2 <= y)
+end
+
+function prims.destroy(t)
+    windower.prim.delete(meta[t].name)
+    meta[t] = nil
+    t = nil
+end
+
+function prims.get_events(t)
+    return meta[t].events
+end
+
+function prims.register_event(t, event, fn)
+    if not events[event] then
+        error('The event ' .. event .. ' is not available to the ' .. class(t) .. ' class.')
+        return
+    end
+    
+    local m = meta[t].events
+
+    m[event] = m[event] or {n = 0}
+    local n = #m[event] + 1
+    m[event][n] = fn
+    m[event].n = m[event].n > n and m[event].n or n
+
+    return n
+end
+ 
+function prims.unregister_event(t, event, n)
+    if not (events[event] and meta[t].events[event]) then
+        return
+    end
+
+    if type(n) == 'number' then
+        meta[t].events[event][n] = nil
+    else
+        for i = 1, meta[t].events[event].n do
+            if meta[t].events[event][i] == n then
+                meta[t].events[event][i] = nil
+                return
+            end
+        end
+    end
+end
+
+return prims
```

Comment:
Oh I didn't realize they were different, they looked very similar superficially. I'll merge this now, can swap it out (and update the existing library if you feel like it) at another point.